llvm/lib/Target/DirectX/DXILOpBuilder.cpp

```diff
@@ -450,6 +462,10 @@ CallInst *DXILOpBuilder::createOp(dxil::OpCode OpCode, ArrayRef<Value *> Args,
   return *Result;
 }
 
+StructType *DXILOpBuilder::getResRetType(Type *ElementTy) {
+  return ::getResRetType(ElementTy);
```

Comment:
Trying to understand this pattern...it being a non-static member function would strongly imply to me as the caller that this function somehow depends on the internal state of DXILOpBuilder.  But it doesn't.  Do you think it will in the future?  Or is there some other design consideration in play?

Reply:
This pattern isn't great, and I expect it to go away eventually. The `getResRetType` static function already exists and is called by other static functions, so simply making it a member involves quite a bit of code churn. Further, most of the uses in static functions are in code that has various TODOs about refactoring how we use TableGen in order to simplify it, so I'd rather not spend a lot of time on it until we actually do that.